### ISSUE TITLE
Same email sent twice

### ISSUE THREAD

**Issue by chandon** [mautic/mautic#2215]
What type of report is this:

| Q | A |
| --- | --- |
| Bug report? | Y |
| Feature request? |  |
| Enhancement? |  |
## Description:

I'm using swift with "queue mode" to send email.
I send my emails with a cron task, every minute 

```
php app/console mautic:emails:send
```

Some of my emails are sent twice (i mean that the same contact receive the same message twice, with the same messageId in the header. I can also confirm that 2 emails are sent through the log of my postfix SMTP server, by 2 different process, at the same time)

I think that sending a large bunch of email make the first **mautic:emails:send** last more than 1 minute (even more than 5 minutes...). So 1 minute later, i must have a second **mautic:emails:send** process.
So, at the same time, i've got 2 processes sending my emails, perhaps accessing at the same time the files in my app/spool/default (swift email queue)

I think there's a lack of **mautic:emails:send** anti-concurrency. The anti-concurrency could be
- per email (best for cluster architecture, but more difficult to implement)
- per process (do not lauch process twice, but not enough for cluster architecture)
## If a bug:

| Q | A |
| --- | --- |
| Mautic version | 2.0.2-dev (2016-08-08) |
| PHP version | 5.6.19 |
### Steps to reproduce:
1. Create a campaign sending an email
2. Configure your email through swift in "queue mode" (but do not configure the cron)
3. Add a large bunch of contact to your campaign
4. Launch **mautic:emails:send** twice in parallel fro your console
5. See some emails been sent twice

**Comment by chandon** [comment]
In the DB, table email_stats, "retry_count" = 1 for each mail send twice

**Comment by escopecz** [comment]
How do you "Launch mautic:emails:send twice in parallel fro your console"? There is a mechanism to avoid running multiple commands in parallel. You'd have to use the `--force` parameter. Is that what you do?

**Comment by chandon** [comment]
Cron does launch it twice (the second process 1 minute later the first). For test purpose, you can do it from the console. I don't run my command with --force parameter

To simulate and run in parallel from your console, write a script 

    #!/bin/bash
    php app/console mautic:emails:send &
    php app/console mautic:emails:send &

If you want, you can even add a sleep(50); in the middle of the **ProcessEmailQueueCommand** class. There is no message, no error, both script run for 50s... So i'm not sure the "mechanism" is working well

Btw, i'm not 100% certain its the problem for twice sending, by i don't see other possible way for this bug...

**Comment by chandon** [comment]
And the is no --force option for **mautic:emails:send**

**Comment by escopecz** [comment]
I think this is duplicate of https://github.com/mautic/mautic/issues/1737. I'll close this issue. There are some interesting theories about how to handle the parallel processes better.

**Comment by chandon** [comment]
#2217 only working for 1 single computer hosting

**Comment by chandon** [comment]
but swift pool too...

**Comment by escopecz** [comment]
Fixed in 2.3
